resources/migrations/001_update_migrations.yaml

```
@@ -9270,6 +9270,90 @@ databaseChangeLog:
                   type: ${text.type}
                   remarks: running, failed, or completed
 
+
+  - changeSet:
+      id: v51.2024-08-28T07:03:41
+      author: johnswanson
+      comment: Add `report_card.dashboard_id`
+      changes:
+        - addColumn:
+            tableName: report_card
+            columns:
+              - column:
+                  name: dashboard_id
+                  type: int
+                  remarks: The dashboard that owns the card, if it is a dashboard-internal card.
+                  constraints:
+                    nullable: true
+
+  - changeSet:
+      id: v51.2024-08-28T07:03:42
+      author: johnswanson
+      comment: Add `report_dashboardcard.is_dashboard_internal_card`
+      changes:
+        - addColumn:
+            tableName: report_dashboardcard
+            columns:
+              - column:
+                  name: is_dashboard_internal_card
+                  type: ${boolean.type}
+                  remarks: |
+                    A denormalized representation that this edge from dashboard<->card should be the *only* edge from its card.
+                    Provides an efficient way to enforce the fact that
```

Watch out for this trailing sentence. How about the following?

```suggestion
                    A denormalized edge from dashboard<->card which uses a unique index to ensure that
                    there is at most one `is_dashboard_internal_card` dashcard per card.
```

This also seems to be attached to the wrong changeset, and belongs on the next one.

---

I ended up deciding against creating this column at all, as it was confusing and painful to keep up-to-date and didn't provide a lot of value (e.g. as you pointed out, if we move the dashboard card to a new dashboard or a collection, we'd need to update the dashcards as well).